django/db/backends/mysql/operations.py

```diff
@@ -294,12 +294,18 @@ def subtract_temporals(self, internal_type, lhs, rhs):
 
     def explain_query_prefix(self, format=None, **options):
         # Alias MySQL's TRADITIONAL to TEXT for consistency with other backends.
+        # If TREE is supported, use that by default as it is more informative.
         if format and format.upper() == 'TEXT':
             format = 'TRADITIONAL'
+        elif not format and 'TREE' in self.connection.features.supported_explain_formats:
+            format = 'TREE'
+        analyze = bool(options.pop('analyze', False))
         prefix = super().explain_query_prefix(format, **options)
-        if format:
+        if analyze and self.connection.features.needs_analyze_statement:
+            prefix = 'ANALYZE'
+        if format and (not analyze or self.connection.features.needs_analyze_statement):
```

Comment:
I don't understand this change :thinking: `analyze` is also untested on MySQL and MariaDB.

Reply:
MySQL 8.0.18 added `EXPLAIN ANALYZE`. It does not accept the `FORMAT=...` option with the `ANALYZE` option, and `EXPLAIN ANALYZE` always uses the new `TREE` format added in 8.0.16.

MariaDB had `EXPLAIN ANALYZE` prior to 10.1.0 whereupon it was changed to be a separate `ANALYZE` statement. This separate statement accepts the `FORMAT=` option.

The change here was to only set the `FORMAT=` option if it is specified, and if we're not using the `ANALYZE` option (to exclude for MySQL) or we're using the `ANALYZE` statement (to include for MariaDB).

You're right - this needs some tests.

Reply:
So maybe (it's clearer to me):
```python
if format:
    # Only MariaDB supports the analyze option with formats.
    if analyze and not connection.mysql_is_mariadb):
        raise ....
```

Reply:
Hmm, that would then cause an exception to be raised for MySQL where we are now setting the format to `'TREE'` by default. I'd like to avoid throwing the exception if possible, like we do with `EXTENDED` where we just ignore it.